File: frontend/src/features/types/court/Court.ts
import type { ApiResponse, GeneralStatusType } from "@/types";
import type { Sport } from "../sport/Sport";

export const CourtSurface = {
    HARD: "HARD",
    CLAY: "CLAY",
    GRASS: "GRASS",
    SYNTHETIC: "SYNTHETIC",
    WOOD: "WOOD",
    OTHER: "OTHER",
} as const;

export type CourtSurfaceType = (typeof CourtSurface)[keyof typeof CourtSurface];

export interface Court {
    name: string;
    slug: string;
    locationDetails: string;
    imgUrl: string;
    priceH: number;
    capacity: number;
    isIndoor: boolean;
    surface: CourtSurfaceType;
    sports: Sport[];
    status: GeneralStatusType;
    isActive: boolean;
}

export type CourtResponse = ApiResponse<Court>;

File: backend/fastapi/app/schemas/court.py
from pydantic import BaseModel
from typing import Optional

class CourtResponse(BaseModel):
    slug: str
    name: str
    location_details: Optional[str]
    img_url: Optional[str]
    price_h: float
    capacity: int
    is_indoor: bool
    surface: str
    status: str
    is_active: bool


class CourtFilteredResponse(BaseModel):
    min_price: Optional[float]
    max_price: Optional[float]
    min_capacity: Optional[int]
    max_capacity: Optional[int]
